### PR TITLE
Add swagger docs

### PR DESCRIPTION
**Pull request by alinagoal**
Created swagger docs for:
- user
- userrole
- auth
- roles

### CODE REVIEW

File: apps/server/src/user/user.controller.ts [hunk 1]
```diff
@@ -10,11 +10,18 @@ import {
...
   ParseIntPipe,
 } from '@nestjs/common';
-import { ApiBearerAuth } from '@nestjs/swagger';
+
+import {
+  ApiBearerAuth,
+  ApiTags,
+  ApiCreatedResponse,
+  ApiBadRequestResponse,
+} from '@nestjs/swagger';
```
Comment (yachujoshi-lf) [inline apps/server/src/user/user.controller.ts:19]: 🪜

File: apps/server/src/user/user.controller.ts [hunk 2]
```diff
@@ -10,11 +10,18 @@ import {
...
 
 import { UserService } from '@/user/user.service';
 
 import { Roles } from '@/decorator/roles.decorator';
+import { User } from './entities/user.entity';
```
Comment (yachujoshi-lf) [inline apps/server/src/user/user.controller.ts:24]: Import alias

File: apps/server/src/user/user.controller.ts [hunk 3]
```diff
@@ -26,6 +33,11 @@ export class UserController {
   constructor(private readonly userService: UserService) {}
 
   @Post()
+  @ApiCreatedResponse({
+    description: 'Created a new user',
+    type: User,
+  })
+  @ApiBadRequestResponse({ description: 'User cannot be created' })
   @HttpCode(HttpStatus.CREATED)
```
Comment (yachujoshi-lf) [inline apps/server/src/user/user.controller.ts:41]: No need for `HttpCode`.

File: apps/server/src/auth/auth.controller.ts [hunk 4]
```diff
@@ -1,4 +1,17 @@
...
+} from '@nestjs/swagger';
+
+import {
+  Get,
+  Request,
+  HttpCode,
+  UseGuards,
+  Controller,
+  HttpStatus,
+} from '@nestjs/common';
```
Comment (yachujoshi-lf) [inline apps/server/src/auth/auth.controller.ts:14]: This would be at the top

File: apps/server/src/auth/auth.controller.ts [hunk 5]
```diff
@@ -18,12 +32,19 @@ export class AuthController {
   async googleAuth() {}
 
   @Get('redirect')
+  @ApiCreatedResponse({
+    description: 'Redirected',
+  })
+  @ApiBadRequestResponse({ description: 'Redirection failed' })
+  @HttpCode(HttpStatus.OK)
```
Comment (yachujoshi-lf) [inline apps/server/src/auth/auth.controller.ts:39]: Not required.

File: apps/server/src/auth/auth.controller.ts [hunk 6]
```diff
@@ -18,12 +32,19 @@ export class AuthController {
...
   @UseGuards(AuthenticationGuard)
   authenticationRedirect(@Request() req: HttpRequest) {
     return this.authService.requestAuthentication(req);
   }
 
   @Get('google/callback')
+  @ApiBadRequestResponse({ description: 'Redirection failed' })
+  @HttpCode(HttpStatus.OK)
```
Comment (yachujoshi-lf) [inline apps/server/src/auth/auth.controller.ts:47]: .

File: apps/server/src/user/user.controller.ts (hunk 1 above)
Comment (alinagoal) [reply apps/server/src/user/user.controller.ts]: Done

File: apps/server/src/user/user.controller.ts (hunk 3 above)
Comment (alinagoal) [reply apps/server/src/user/user.controller.ts]: Done

File: apps/server/src/auth/auth.controller.ts (hunk 5 above)
Comment (alinagoal) [reply apps/server/src/auth/auth.controller.ts:39]: Done

File: apps/server/src/auth/auth.controller.ts (hunk 6 above)
Comment (alinagoal) [reply apps/server/src/auth/auth.controller.ts:47]: Done

File: apps/server/src/user/user.controller.ts (hunk 2 above)
Comment (alinagoal) [reply apps/server/src/user/user.controller.ts]: Done

File: apps/server/src/user/user.controller.ts [hunk 7]
```diff
@@ -1,50 +1,74 @@
...
   constructor(private readonly userService: UserService) {}
 
   @Post()
-  @HttpCode(HttpStatus.CREATED)
+  @ApiCreatedResponse({
+    description: 'Created a new user',
+    type: User,
+  })
+  @ApiBadRequestResponse({ description: 'User cannot be deleted' })
+  @ApiBadRequestResponse({ description: 'User cannot be created' })
```
Comment (yachujoshi-lf) [inline apps/server/src/user/user.controller.ts:41]: Duplicate code

Comment (yachujoshi-lf) [reply apps/server/src/user/user.controller.ts:41]: Can combine in a single statement

**Review by yachujoshi-lf — APPROVED** [review]

File: apps/server/src/user/user.controller.ts (hunk 7 above)
Comment (alinagoal) [reply apps/server/src/user/user.controller.ts:41]: Done

File: apps/server/src/filter/jwt-auth.guard.ts [hunk 8]
```diff
@@ -1,20 +1,23 @@
...
 import { ROLES_KEY } from '@decorator/roles.decorator';
 import { SKIP_AUTH_KEY } from '@/decorator/skip-auth.decorator';
 
 import { RoleEnum } from '@/common/constants/role.enum';
 import { ErrorMessage } from '@common/errors/error.message';
+
 import { RestException } from '@common/exceptions/rest.exception';
+
 import { ErrorDescription } from '@common/errors/constants/description.error';
+import { User } from '@/user/entities/user.entity';
```
Comment (SamipSthaLF) [inline apps/server/src/filter/jwt-auth.guard.ts:20]: 🪜

File: apps/server/src/user/dto/create-user.dto.ts [hunk 9]
```diff
@@ -4,7 +4,7 @@ import { IsString } from 'class-validator';
 
 export class CreateUserDto {
   @IsString()
-  @ApiProperty()
+  @ApiProperty({ default: 'lftechnology.com' })
```
Comment (SamipSthaLF) [inline apps/server/src/user/dto/create-user.dto.ts:7]: lets not have leapfrog.com lets have user@mail.com

File: apps/server/src/user/entities/user.entity.ts [hunk 10]
```diff
@@ -9,17 +9,21 @@ import {
...
 } from 'typeorm';
 
+import { ApiProperty } from '@nestjs/swagger';
+
 @Entity()
 export class User extends BaseEntity {
   @PrimaryGeneratedColumn()
   id: number;
 
+  @ApiProperty({ default: 'lftech@gmail.com' })
```
Comment (SamipSthaLF) [inline apps/server/src/user/entities/user.entity.ts:19]: .

File: apps/server/src/user/user.controller.ts [hunk 11]
```diff
@@ -1,50 +1,73 @@
...
+  ApiBearerAuth,
+  ApiOkResponse,
+  ApiCreatedResponse,
+  ApiBadRequestResponse,
+} from '@nestjs/swagger';
 
 import { UserService } from '@/user/user.service';
 
 import { Roles } from '@/decorator/roles.decorator';
+import { User } from '@/user/entities/user.entity';
```
Comment (SamipSthaLF) [inline apps/server/src/user/user.controller.ts:23]: 🪜

**Review by SamipSthaLF — APPROVED** [review]

File: apps/server/src/filter/jwt-auth.guard.ts (hunk 8 above)
Comment (alinagoal) [reply apps/server/src/filter/jwt-auth.guard.ts:20]: Done

File: apps/server/src/user/dto/create-user.dto.ts (hunk 9 above)
Comment (alinagoal) [reply apps/server/src/user/dto/create-user.dto.ts:7]: Done

File: apps/server/src/user/entities/user.entity.ts (hunk 10 above)
Comment (alinagoal) [reply apps/server/src/user/entities/user.entity.ts:19]: Done

File: apps/server/src/user/user.controller.ts (hunk 11 above)
Comment (alinagoal) [reply apps/server/src/user/user.controller.ts:23]: Done

**Pull request merged**